src/org/mozilla/javascript/NativeArray.java

```diff
@@ -1962,6 +1970,19 @@ private static Object js_copyWithin(
         return thisObj;
     }
 
+    private static Object js_at(Context cx, Scriptable scope, Scriptable thisObj, Object[] args) {
+        Scriptable o = ScriptRuntime.toObject(cx, scope, thisObj);
+        Object targetArg = (args.length >= 1) ? args[0] : Undefined.instance;
+        long relativeIndex = (long) ScriptRuntime.toInteger(targetArg);
+        long len = getLengthProperty(cx, o);    
+        long k = (relativeIndex >= 0) ? relativeIndex : len + relativeIndex;
+        if ((k < 0) || (k >= len)) {
+            return Undefined.instance;
+        }
+
+        return ScriptableObject.getProperty(thisObj, (int) k);
```

Comment:
Thank you for your fix.

I think it is better to use `getElem` or `getRawElem`.
https://github.com/mozilla/rhino/blob/c4c10f4f046fd72c9671636cb9a504f5f6b32953/src/org/mozilla/javascript/NativeArray.java#L990-L1001

And can you add a test for the ArrayLike?

Reply:
With the current code, this is the result.
```js
var arrayLike = {
  0x80000000: 'Integer.MAX_VALUE + 1',
  length: 0x80000001  // Integer.MAX_VALUE + 2
};
console.log(Array.prototype.at.call(arrayLike, 0x80000000)); // expected 'Integer.MAX_VALUE + 1', but actual undefined
```